Support starting log tailers at the end of file (#5165)
grafana/agent: docs/sources/flow/reference/components/loki.source.file.md

```

## Arguments

The component starts a new reader for each of the given `targets` and fans out
log entries to the list of receivers passed in `forward_to`.

`loki.source.file` supports the following arguments:

| Name             | Type                 | Description                                                                       | Default | Required |
| ---------------- | -------------------- | --------------------------------------------------------------------------------- | ------- | -------- |
| `targets`        | `list(map(string))`  | List of files to read from.                                                       |         | yes      |
| `forward_to`     | `list(LogsReceiver)` | List of receivers to send log entries to.                                         |         | yes      |
| `encoding`       | `string`             | The encoding to convert from when reading files.                                  | `""`    | no       |

> | `start_from_eof` | `bool`               | Whether the file should be tailed from the end if a cached position is not found. | `false` | no       |

Reviewer: ```suggestion
| `start_from_eof` | `bool`               | Whether a log file should be tailed from the end if a cached position is not found. | `false` | no       |
```

I assume there could be more than one log file?

The `encoding` argument must be a valid [IANA encoding][] name. If not set, it
defaults to UTF-8.

> The `start_from_eof` argument is valuable when you want to tail a large file without reading its entire content.

Reviewer: ```suggestion
You can use the `start_from_eof` argument when you want to tail a large file without reading its entire content.
```

When set to true, only new logs will be read, ignoring the existing ones.

> 

Author: I formatted this document in a specific commit, I hope this does not make the review too annoying. I find the doc easier to read/modify when it is nicely formatted,

```

## Arguments

The component starts a new reader for each of the given `targets` and fans out
log entries to the list of receivers passed in `forward_to`.

`loki.source.file` supports the following arguments:

| Name            | Type                 | Description                                                                         | Default | Required |
| --------------- | -------------------- | ----------------------------------------------------------------------------------- | ------- | -------- |
| `targets`       | `list(map(string))`  | List of files to read from.                                                         |         | yes      |
| `forward_to`    | `list(LogsReceiver)` | List of receivers to send log entries to.                                           |         | yes      |
| `encoding`      | `string`             | The encoding to convert from when reading files.                                    | `""`    | no       |

> | `tail_from_end` | `bool`               | Whether a log file should be tailed from the end if a cached position is not found. | `false` | no       |

Reviewer 2: Will users know what 'cached position' means? This is the first mention of it in this page. We should probably use similar vocabulary like the one in 'component behaviour' section.

Author: changed to 'stored' to match the vocabulary of the section in the component behavior

## Debug information

`loki.source.file` exposes some target-level debug information per reader:

- The tailed path.
- Whether the reader is currently running.
- What is the last recorded read offset in the positions file.

## Debug metrics

- `loki_source_file_read_bytes_total` (gauge): Number of bytes read.
- `loki_source_file_file_bytes_total` (gauge): Number of bytes total.
- `loki_source_file_read_lines_total` (counter): Number of lines read.
- `loki_source_file_encoding_failures_total` (counter): Number of encoding failures.
- `loki_source_file_files_active_total` (gauge): Number of active files.

> ## Component behavior

Reviewer 2: I think that this section should be amended to mention the option that you can tail?
